Content.Shared/Light/LightLevelSystem.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Linq;
+using Content.Shared.Physics;
+using JetBrains.Annotations;
+using Robust.Shared.Map;
+using Robust.Shared.Physics;
+using Robust.Shared.Physics.Systems;
+
+namespace Content.Shared.Light;
+
+public sealed class LightLevelSystem : EntitySystem
+{
+    [Dependency] private readonly SharedPhysicsSystem _physics = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+
+    private bool Ignored(EntityUid ignored)
+    {
+        return !Transform(ignored).Anchored;
+    }
+
+    [PublicAPI]
+    public float GetLightLevel(EntityUid uid, TransformComponent? xform = null)
+    {
+        if (!Resolve(uid, ref xform))
+            return 0f;
+        var pos = _transform.GetWorldPosition(xform);
+        return GetLightLevel(pos, xform.MapID);
+    }
+
+    [PublicAPI]
+    public float GetLightLevel(Vector2 pos, MapId map)
+    {
+        // todo: in the future, this should support light on planet maps.
+        var lightLevel = 0f;
+
+        foreach (var (light, xform) in EntityQuery<SharedPointLightComponent, TransformComponent>(true))
```

Comment:
surely there are enough lights on the map that doing an entitylookup is faster than a query + distance check